README.md

```diff
@@ -81,7 +87,152 @@ credentials for third-party services manually in order to run the development
 server or the integration tests.
 
 The environment variables in `.env` customize the settings from
-`opencraft/settings.py` which are loaded via `env()`.
+`opencraft/settings.py` which are loaded via `env()`. For more information about
+each setting, see [`opencraft/settings.py`](opencraft/settings.py). As a
+minimum, you will need to add credentials for OpenStack, Gandi and GitHub. You
+can use this example `.env` file as a starting point:
+
+```sh
+SECRET_KEY='...'
+DATABASE_URL='postgres://localhost/opencraft'
+OPENSTACK_USER='username'
+OPENSTACK_PASSWORD='password'
+OPENSTACK_TENANT='tenant-name'
+OPENSTACK_AUTH_URL='https://auth.cloud.ovh.net/v2.0'
+OPENSTACK_REGION='BHS1'
+OPENSTACK_SANDBOX_SSH_KEYNAME='keypair-name'
+INSTANCES_BASE_DOMAIN='example.com'
+GANDI_ZONE_ID='123456789'
+GANDI_API_KEY='api-key'
+GITHUB_ACCESS_TOKEN='github-token'
+WATCH_ORGANIZATION='github-org'
+```
+
+### A note on SSH keys
+
+The instance manager uses [Ansible](https://www.ansible.com/) to provision
+openedx sandboxes. Ansible uses SSH to run commands on remote servers, so you
+will need to configure the instance manager with an SSH key pair to use for this
+purpose.
+
+First, **as the user that the instance manager runs as**, run:
+
+    ssh-keygen -t rsa -b 4096
+
+This will create an ssh key pair for that user, saving it at `~/.ssh/id_rsa` and
+`~/.ssh/id_rsa.pub` by default. Next, we need to upload the public key to
+OpenStack. Make sure the nova command line client is installed:
+
+    pip install python-novaclient
+
+[Configure the client with your OpenStack credentials](http://docs.openstack.org/cli-reference/common/cli_set_environment_variables_using_openstack_rc.html),
+then run:
+
+    nova keypair-add --pub_key ~/.ssh/id_rsa.pub KEY_NAME
+
+where `KEY_NAME` is the name used to identify this key pair in OpenStack. The
+`OPENSTACK_SANDBOX_SSH_KEYNAME` setting in your `.env` file should be set to
+this name.
+
+### OpenStack images
+
+Open edX is currently designed to run on Ubuntu 12.04. Your OpenStack host may
+already have an image available for this version of Ubuntu, but for maximum
+compatibility we recommend the
+[official Ubuntu cloud image](https://cloud-images.ubuntu.com/precise/current/).
+To add this image to OpenStack, install glance:
+
+    pip install python-glanceclient
+
+Then, fetch the image and add it to OpenStack:
+
+    wget https://cloud-images.ubuntu.com/precise/current/precise-server-cloudimg-amd64-disk1.img
+    glance image-create \
+      --disk-format=qcow2 \
+      --container-format=bare \
+      --file precise-server-cloudimg-amd64-disk1.img \
+      --name IMAGE_NAME \
+      --progress
+
+where `IMAGE_NAME` is the name used to identify the image in OpenStack. The
+`OPENSTACK_SANDBOX_BASE_IMAGE` setting in your `.env` file should match this
+name:
+
+    OPENSTACK_SANDBOX_BASE_IMAGE='{"name": "IMAGE_NAME"}'
+
+### OpenStack flavours
```

Comment:
@omarkhan Bitten by British English..  I know that's really how flavours is spelled, but it should probably be in American like the rest of the document :)